packages/client-beta/src/services/sounds-filtering.ts

```diff
@@ -0,0 +1,24 @@
+import { XSound } from '@service/domain/x-sounds/x-sound';
+
+export function soundMatchesPhrase(sound: XSound, phrase: string) {
+  const { name, tags = [] } = sound;
+  const tagsAndNameJoinedString = [name, ...tags].join(' ').toLowerCase();
+  return phrase
+    .toLowerCase()
+    .split(' ')
+    .every((keyword) => tagsAndNameJoinedString.includes(keyword));
+}
+
+export function soundsSorting(phrase: string) {
+  return function compare(a: XSound, b: XSound) {
+    const isGoodMatchA = a.name.startsWith(phrase);
+    const isGoodMatchB = b.name.startsWith(phrase);
+
+    if (isGoodMatchA !== isGoodMatchB) {
+      return isGoodMatchA ? -1 : 1;
+    }
+
+    return a.name.localeCompare(b.name, undefined, { sensitivity: 'base' }) || b.timesPlayed - a.timesPlayed;
+  };
+}
+
```

Comment:
I'm not saying anything, I'm not suggesting anything - those methods looks like a great opportunity to write some tests 😉

Reply:
Thanks to @gorzalczany for these beautifully filtering features :D I copied them from a previous soundboard